### PR TITLE
 feat: External Private Solicitors creates cases on behalf of local authority(FPLA-2604)

### PR DESCRIPTION
**Pull request by tomaszpowroznik**
https://github.com/hmcts/fpla-docker/pull/51
https://github.com/hmcts/fpla-docker/pull/52
https://github.com/hmcts/fpl-ccd-configuration/pull/2197
https://github.com/hmcts/fpl-ccd-configuration/pull/2235

org
users

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseInitiationService.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                    .getLocalAuthorityName(localAuthorityService.getLocalAuthorityCode()))
+                .build();
+        }
+    }
+
+    public void grantCaseAccess(CaseData caseData) {
+        final Long caseId = caseData.getId();
+
+        if (nonNull(caseData.getOutsourcingPolicy())) {
+            caseAccessService.revokeCaseRoleFromUser(caseId, requestData.userId(), CREATOR);
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseInitiationService.java:99]: curiosity: did we find out why the CREATOR role is an issue and has to be removed?

Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseInitiationService.java:99]: yeah, if there is single user who has access to a case and have CREATOR role then supplementary data is not populated and sharing a case is not possible

https://tools.hmcts.net/confluence/display/RCCD/A+Guide+to+Assign+Access+to+cases+for+professional+users%3A+configuration

File: ccd-definition/AuthorisationCaseField/CareSupervision/eps-managing.json [hunk 2]
```diff
@@ -0,0 +1,1024 @@
+[
```
Comment (samuelzerozero) [inline ccd-definition/AuthorisationCaseField/CareSupervision/eps-managing.json:1]: matches 1 to 1 to la-solicitor.json apart from the generic fields assigned to caseworker-publiclaw-solicitor ✅

**Review by samuelzerozero — DISMISSED** [review]

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/DynamicListHelper.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.fpl.service.CaseConverter.MAP_TYPE;
+
+@Service
+public class DynamicListHelper {
```
[the remark this answers is not in the document]
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/testingsupport/DynamicListHelper.java:18]: this is test util only, non production code, dont think that test is required

File: service/src/main/java/uk/gov/hmcts/reform/ccd/model/OrganisationPolicy.java [hunk 4]
```diff
@@ -19,4 +20,18 @@
     @JsonProperty("OrgPolicyCaseAssignedRole")
     private String orgPolicyCaseAssignedRole;
 
+    public static OrganisationPolicy organisationPolicy(String organisationId, CaseRole caseRole) {
+
+        if (organisationId == null) {
```
[the remark this answers is not in the document]
Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/ccd/model/OrganisationPolicy.java:25]: i agree, returning optional is most preferable option here. Dont like exception approach, as this service should not make decision if this is an exception situation or not, client of this api should make such decision

**Pull request merged**